app/models/managed_interactive.rb

```diff
@@ -0,0 +1,171 @@
+  end
+
+  def enable_learner_state
+    library_interactive && library_interactive ? library_interactive.enable_learner_state : false
+  end
+
+  def show_delete_data_button
+    library_interactive && library_interactive ? library_interactive.show_delete_data_button : false
+  end
+
+  def self.string_name
+    "managed interactive"
+  end
+
+  def self.portal_type
+    "iframe interactive"
+  end
+
+  def to_hash
+    # Deliberately ignoring user (will be set in duplicate)
+    {
+      library_interactive_id: library_interactive_id,
+      name: name,
+      url_fragment: url_fragment,
+      authored_state: authored_state,
+      is_hidden: is_hidden,
+      is_full_width: is_full_width,
+      show_in_featured_question_report: show_in_featured_question_report,
+      inherit_aspect_ratio_method: inherit_aspect_ratio_method,
+      custom_aspect_ratio_method: custom_aspect_ratio_method,
+      inherit_native_width: inherit_native_width,
+      custom_native_width: custom_native_width,
+      inherit_native_height: inherit_native_height,
+      custom_native_height: custom_native_height,
+      inherit_click_to_play: inherit_click_to_play,
+      custom_click_to_play: custom_click_to_play,
+      inherit_full_window: inherit_full_window,
+      custom_full_window: custom_full_window,
+      inherit_click_to_play_prompt: inherit_click_to_play_prompt,
+      custom_click_to_play_prompt: custom_click_to_play_prompt,
+      inherit_image_url: inherit_image_url,
+      custom_image_url: custom_image_url
+    }
+  end
+
+  def duplicate
+    # Generate a new object with those values
+    ManagedInteractive.new(self.to_hash)
+    # N.B. the duplicate hasn't been saved yet
+  end
+
+  def export
+    hash = to_hash()
+    hash.delete(:library_interactive_id)
+    hash[:library_interactive] = {
+      hash: library_interactive.generate_export_hash(),
+      data: library_interactive.to_hash()
+    }
+    hash.to_json()
+  end
+
+  def self.import(import_hash)
+    # save off the imported library interactive to hydrate it after the instance is created
+    parsed = JSON.parse(import_hash)
+    imported_library_interactive = parsed["library_interactive"]
+    parsed.delete("library_interactive")
+
+    managed_interactive = self.new(parsed)
+
+    # find the existing matching library interactive or create a new one
+    if imported_library_interactive
+      library_interactive = LibraryInteractive.find_by_export_hash(imported_library_interactive["hash"])
+      if library_interactive
+        managed_interactive.library_interactive = library_interactive
+      else
+        managed_interactive.library_interactive = LibraryInteractive.import(imported_library_interactive["data"])
+
+        # REVIEW QUESTION: should we call #save here - if we don't and the user imports an activity
+        # that uses the same library interactive more than once it won't find the newly created
+        # library interactive
```

Comment:
Go ahead with the save.  Just to know about it it would be good to look at the second pass code that is handling related linked interactives. And so we don't forget about this save when allow authors to import, add a comment to the top level import.